cmd/clusterctl/client/repository/repository_github_test.go

```diff
@@ -185,54 +187,47 @@ func Test_gitHubRepository_getReleaseByTag(t *testing.T) {
 	tests := []struct {
 		name        string
 		args        args
-		wantTagName string
+		wantTagName *string
 		wantErr     bool
 	}{
 		{
 			name: "Return existing version",
 			args: args{
 				tag: "foo",
 			},
-			wantTagName: "v0.4.1",
+			wantTagName: pointer.StringPtr("v0.4.1"),
 			wantErr:     false,
 		},
 		{
 			name: "Fails if version does not exists",
 			args: args{
 				tag: "bar",
 			},
-			wantTagName: "",
+			wantTagName: nil,
 			wantErr:     true,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			g, err := newGitHubRepository(providerConfig, configVariablesClient)
-			if err != nil {
-				t.Fatal(err)
-			}
-			g.injectClient = client
+			gRepo, err := newGitHubRepository(providerConfig, configVariablesClient)
+			g.Expect(err).NotTo(HaveOccurred())
 
-			got, err := g.getReleaseByTag(tt.args.tag)
-			if (err != nil) != tt.wantErr {
-				t.Fatalf("error = %v, wantErr %v", err, tt.wantErr)
-			}
+			gRepo.injectClient = client
+
+			got, err := gRepo.getReleaseByTag(tt.args.tag)
 			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
 				return
 			}
 
-			if tt.wantTagName == "" && got == nil {
-				return
-			}
+			g.Expect(err).NotTo(HaveOccurred())
```

Comment:
nit we should be consistent in having assertion on the operation err/on wantErr like in Test_componentsClient_Get